### ISSUE TITLE
NullPointerException on  response.setContentLength(0); response.getOutputStream().print("");

### ISSUE THREAD

**Issue by bio-boris** [eclipse/jetty.project#3545]
The following code produces a NPE
```
java.lang.NullPointerException
	at org.eclipse.jetty.server.HttpOutput.print(HttpOutput.java:734)
	at org.eclipse.jetty.server.HttpOutput.print(HttpOutput.java:689)
```

 when overriding the doOptions method in the  `HttpServlet` for jetty `9.4.15.v20190215` but works in `9.4.12.v20180830`  

```
@Override
protected void doOptions(HttpServletRequest request, HttpServletResponse response)
        throws ServletException, IOException {
    response.setContentLength(0);
    response.getOutputStream().print("");
    response.getOutputStream().flush();
}
```
The fix is to remove `response.getOutputStream().print("");`

**Comment by joakime** [comment]
https://github.com/eclipse/jetty.project/blob/eb70b240169fcf1abbd86af36482d1c49826fa0b/jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java#L718-L734

Wow, yup, that's a legit NPE.

**Comment by joakime** [comment]
Looks like this is a side effect of commit 225760c2de0dff2f9f108bff0541e088af96e510 from Issue #3207

**Comment by joakime** [comment]
I have been able to replicate this in a test case ...

``` java
    @Test
    public void testPrint_Empty() throws Exception
    {
        Response response = getResponse();
        response.setCharacterEncoding(UTF_8.name());

        try(ServletOutputStream outputStream = response.getOutputStream())
        {
            outputStream.print("ABC");
            outputStream.print("");
            outputStream.println();
            outputStream.flush();
        }

        String expected = "ABC\r\n";
        assertEquals(expected,BufferUtil.toString(_content, UTF_8));
    }
```

Results in ...

```
java.lang.NullPointerException
	at org.eclipse.jetty.server.HttpOutput.print(HttpOutput.java:734)
	at org.eclipse.jetty.server.HttpOutput.print(HttpOutput.java:689)
	at org.eclipse.jetty.server.ResponseTest.testPrint_Empty(ResponseTest.java:592)
```

**Comment by joakime** [comment]
Opened PR #3547

**Comment by joakime** [comment]
Fixed in `jetty-9.4.x`.
